packages/semver/src/executors/version/version.ts

```diff
@@ -137,14 +141,20 @@ export function _runStandardVersion({
   bumpFiles: string[];
   skipChangelog: boolean;
 } & CommonVersionOptions) {
-  return standardVersion({
+  const changeLogPath = getChangelogPath(projectRoot);
+
+  const currentChangeLog = await firstValueFrom(
+    readFileIfExists(changeLogPath, changelogHeader)
+  );
+
+  await standardVersion({
     bumpFiles,
     /* Make sure that we commit the manually generated changelogs that
      * we staged. */
     commitAll: true,
     dryRun,
     header: changelogHeader,
-    infile: getChangelogPath(projectRoot),
+    infile: changeLogPath,
```

Comment:
Rather than using the `diff` perhaps capturing the results via a `Buffer` then saving the file outside of the scope of `standardVersion` would be more reliable.

Ownership of committing would probably need to be taken over as well as saving the file would likely come after `standardVersion` and all it's hooks had already completed.